File: forensics-btmgen/src/test/java/de/burger/forensics/plugin/scan/ScannerFacadeIntegrationTest.java
// DEST: src/test/java/de/burger/forensics/plugin/scan/ScannerFacadeIntegrationTest.java
package de.burger.forensics.plugin.scan;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;
import org.junit.jupiter.api.Test;

class ScannerFacadeIntegrationTest {

    @Test
    void scansJavaAndKotlinSourcesTogether() throws IOException {
        Path root = Files.createTempDirectory("facade-ast");
        Path javaFile = root.resolve("Joint.java");
        Files.writeString(javaFile, String.join("\n",
                "package com.example;",
                "",
                "public class Joint {",
                "    public void run(boolean ok) {",
                "        if (ok) {",
                "            System.out.println(\"OK\");",
                "        }",
                "    }",
                "}");
        Path kotlinFile = root.resolve("Joint.kt");
        Files.writeString(kotlinFile, String.join("\n",
                "package com.example",
                "",
                "class JointKt {",
                "    fun test(value: Int) = if (value > 0) \"p\" else \"n\"",
                "}");

        List<ScanEvent> events = new ScannerFacade().scan(root, List.of(), List.of());
        assertFalse(events.isEmpty(), "Expected events to be emitted");
        Set<String> languages = events.stream().map(ScanEvent::language).collect(Collectors.toSet());
        assertEquals(Set.of("java", "kotlin"), languages);
        assertTrue(events.stream().anyMatch(event -> "java".equals(event.language()) && "if-true".equals(event.kind())));
        assertTrue(events.stream().anyMatch(event -> "kotlin".equals(event.language()) && "if-true".equals(event.kind())));
    }
}

File: forensics-btmgen/src/test/kotlin/de/burger/forensics/plugin/scan/ScannerFacadeIntegrationTest.kt
package de.burger.forensics.plugin.scan

import java.nio.file.Files
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class ScannerFacadeIntegrationTest {

    @Test
    fun `scans Java and Kotlin sources together`() {
        val root = Files.createTempDirectory("facade-ast")
        val javaFile = root.resolve("Joint.java")
        Files.writeString(
            javaFile,
            """
            package com.example;

            public class Joint {
                public void run(boolean ok) {
                    if (ok) {
                        System.out.println("OK");
                    }
                }
            }
            """.trimIndent()
        )
        val kotlinFile = root.resolve("Joint.kt")
        Files.writeString(
            kotlinFile,
            """
            package com.example

            class JointKt {
                fun test(value: Int) = if (value > 0) "p" else "n"
            }
            """.trimIndent()
        )

        val events = ScannerFacade().scan(root, emptyList(), emptyList())
        assertThat(events).hasSizeGreaterThan(0)
        assertThat(events.map { it.language }.toSet()).containsExactlyInAnyOrder("java", "kotlin")
        assertThat(events).anyMatch { it.language == "java" && it.kind == "if-true" }
        assertThat(events).anyMatch { it.language == "kotlin" && it.kind == "if-true" }
    }
}
